File: lingting-huawei/src/main/java/live/lingting/framework/huawei/iam/HuaweiIamToken.java
package live.lingting.framework.huawei.iam;

import lombok.Getter;
import lombok.Setter;

import java.time.Duration;
import java.time.LocalDateTime;

/**
 * @author lingting 2024-09-12 22:04
 */
@Getter
@Setter
public class HuaweiIamToken {

	private String value;

	private LocalDateTime expire;

	private LocalDateTime issued;

	public HuaweiIamToken() {
	}

	public HuaweiIamToken(String value, LocalDateTime expire, LocalDateTime issued) {
		this.value = value;
		this.expire = expire;
		this.issued = issued;
	}

	public Duration duration(LocalDateTime now) {
		return Duration.between(now, expire);
	}

	public boolean isExpired(Duration tokenEarlyExpire) {
		LocalDateTime now = LocalDateTime.now();
		return isExpired(tokenEarlyExpire, now);
	}

	public boolean isExpired(Duration tokenEarlyExpire, LocalDateTime now) {
		Duration duration = duration(now);
		return duration.compareTo(tokenEarlyExpire) < 1;
	}

}

File: lingting-huawei/src/main/kotlin/live/lingting/framework/huawei/iam/HuaweiIamToken.kt
package live.lingting.framework.huawei.iam

import java.time.Duration

import java.time.LocalDateTime
import live.lingting.framework.time.DateTime

/**
 * @author lingting 2024-09-12 22:04
 */
data class HuaweiIamToken(
    val value: String,
    val expire: LocalDateTime,
    val issued: LocalDateTime,
) {

    fun duration(now: LocalDateTime): Duration {
        return Duration.between(now, expire)
    }

    fun isExpired(tokenEarlyExpire: Duration): Boolean {
        val now = DateTime.current()
        return isExpired(tokenEarlyExpire, now)
    }

    fun isExpired(tokenEarlyExpire: Duration, now: LocalDateTime): Boolean {
        val duration = duration(now)
        return duration.compareTo(tokenEarlyExpire) < 1
    }
}
